### PR TITLE
Add array_replace_keys function

### PR DESCRIPTION
**Pull request by sctape**

### CODE REVIEW

**Comment by sctape** [comment]
@equip/contributors Any thoughts on this PR?

**Comment by elazar** [comment]
I'd remove the `array` prefix, as l don't think we're using that elsewhere in this library. Otherwise,  seems fine.

**Comment by sctape** [comment]
@elazar good suggestion, fixed.

**Comment by shadowhand** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/equip/assist/pull-request/19/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=equip/assist)

File: src/array.php [hunk 1]
```diff
@@ -170,3 +170,20 @@ function index_by($source, $key)
...
+ * @return array
+ */
+function array_replace_keys($source, $keys_to_replace)
+{
+    $source = to_array($source);
+
+    $keys = array_keys($source);
+    $replaced_keys = array_replace(array_combine($keys, $keys), $keys_to_replace);
+
+    return array_combine($replaced_keys, $source);
```
Comment (shadowhand) [inline src/array.php:188]: For anyone wondering, the benefit of this is that it keeps the keys in the same order, whereas a foreach loop and replace would change the order.

**Pull request merged**